src/db.rs

```diff
@@ -149,7 +150,7 @@ pub struct Database {
     mem: TransactionalMemory,
     next_transaction_id: AtomicTransactionId,
     live_read_transactions: Mutex<BTreeSet<TransactionId>>,
-    live_write_transaction: Mutex<Option<TransactionId>>,
+    live_write_transaction: RwLock<Option<TransactionId>>,
```

Comment:
I don't think this needs to change to an `RwLock`. You never call `.read()` and `.lock()` blocks just like `.write()`. Am I missing something?

Reply:
You're totally right. I think maybe I had a prior revision which required a RwLock for some reason.